### ISSUE TITLE
Make text container padding adapt to size

### ISSUE THREAD

**Issue by dwelle** [excalidraw/excalidraw#4498]

**Comment by ad1992** [comment]
I am thinking if we should just reduce padding to `5px` 👇🏻
![Excalidraw (3)](https://user-images.githubusercontent.com/11256141/147817573-bfd53a6e-9f79-4623-a667-ae369911f2d4.gif)

Having different padding depending on size might look a bit inconsistent specially when resizing.

**Comment by zsviczian** [comment]
Even if we set padding to a fix value for now, I would recommend storing padding as a property of the container, maybe as part of boundElements[]?

**Comment by dwelle** [comment]
> I am thinking if we should just reduce padding to 5px

I don't know. For small containers 5px is fine, but for large ones, it won't be pretty.

![image](https://user-images.githubusercontent.com/5153846/147818757-22d0bda6-8728-4dd6-9ca7-cd6f92498772.png)

Also, 5px has an issue where it doesn't account for the actual shape, so the text often overflows:

![image](https://user-images.githubusercontent.com/5153846/147818825-72fac0ba-b20e-439a-a5bd-d0709b42272a.png)

But 10px is already too much for small things like buttons. Not sure how to handle this

> I would recommend storing padding as a property of the container, maybe as part of boundElements[]?

Maybe, but that migration can be done any time as there are no forward compat issue I'm seeing.

**Comment by dwelle** [comment]
Actually, it applies even to `architect` shape with monospace font:

![image](https://user-images.githubusercontent.com/5153846/147818958-e7654e67-ec8c-4762-8392-b2c47b12c7b5.png)

**Comment by ad1992** [comment]
> Actually, it applies even to `architect` shape with monospace font:
> 
> ![image](https://user-images.githubusercontent.com/5153846/147818958-e7654e67-ec8c-4762-8392-b2c47b12c7b5.png)
> 
> It almost seems like the 5px padding is counted from the bounding box selection border, not the actual shape.

Are you resizing the container ? Coz I see this bug when resizing with 5px

**Comment by dwelle** [comment]
> Are you resizing the container ? Coz I see this bug when resizing with 5px

This is 5px, yes (localhost).

**Comment by ad1992** [comment]
> > Are you resizing the container ? Coz I see this bug when resizing with 5px
> 
> This is 5px, yes (localhost).

Yes thats fine, the optimisation for `wrapText` call is not working for 5px so will have to check that
<img width="570" alt="Screenshot 2021-12-31 at 4 18 13 PM" src="https://user-images.githubusercontent.com/11256141/147819192-2a0a8d85-d998-47df-b8b8-5e1c10802e0b.png">
x so I have commented that code in my GIF

**Comment by dwelle** [comment]
Reopening as not implemented yet.

**Comment by codepunkt** [comment]
I'm not sure if this is what I think it is.

I think the very small padding of bounded texts inside their is suboptimal. Understandably, it's hard to decide on the right padding for different sizes of bounded text and their container elements. However, I would like to propose a padding that is based on the font size of the bounded text.

**Comment by JodhwaniMadhur** [comment]
Hello, i am new to this project and wanted to know if this is duplicate or related in some way to #8834 coz I am trying to solve it and I see that changes were made to `BOUND_TEXT_PADDING` variable for the same.
